src/twisted/application/runner/_runner.py

```diff
@@ -46,48 +42,41 @@ def __init__(self, options):
     def run(self):
         """
         Run this command.
-        Equivalent to::
-
-            self.killIfRequested()
-            self.writePIDFile()
-            self.startLogging()
-            self.startReactor()
-            self.reactorExited()
-            self.removePIDFile()
-
-        Additional steps may be added over time, but the order won't change.
         """
+        pidFile = self.options.get(RunnerOptions.pidFile, nonePIDFile)
+
         self.killIfRequested()
-        self.writePIDFile()
-        self.startLogging()
-        self.startReactor()
-        self.reactorExited()
-        self.removePIDFile()
+
+        try:
+            with pidFile:
+                self.startLogging()
+                self.startReactor()
+                self.reactorExited()
+
+        except AlreadyRunningError:
+            exit(ExitStatus.EX_CONFIG, "Already running.")
+            return  # When testing, patched exit doesn't exit
 
 
     def killIfRequested(self):
         """
         Kill a running instance of this application if L{RunnerOptions.kill} is
         specified and L{True} in C{self.options}.
-        This requires that L{RunnerOptions.pidFilePath} also be specified;
+        This requires that L{RunnerOptions.pidFile} also be specified;
         exit with L{ExitStatus.EX_USAGE} if kill is requested with no PID file.
         """
-        pidFilePath = self.options.get(RunnerOptions.pidFilePath)
+        pidFile = self.options.get(RunnerOptions.pidFile)
 
         if self.options.get(RunnerOptions.kill, False):
-            if pidFilePath is None:
-                exit(ExitStatus.EX_USAGE, "No PID file specified")
+            if pidFile is None:
```

Comment:
Presently it could be None, but the next ticket will rework options handling, so let me punt on this issue until then.